Kernel/Time/RTC.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020, Liav A. <liavalb@hotmail.co.il>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <Kernel/Arch/i386/CPU.h>
+#include <Kernel/CMOS.h>
+#include <Kernel/Time/RTC.h>
+#include <Kernel/Time/TimeManagement.h>
+#include <LibBareMetal/IO.h>
+
+namespace Kernel {
+#define IRQ_TIMER 8
+#define MAX_FREQUENCY 8000
+
+NonnullRefPtr<RealTimeClock> RealTimeClock::create(Function<void(const RegisterState&)> callback)
+{
+    return adopt(*new RealTimeClock(move(callback)));
+}
+RealTimeClock::RealTimeClock(Function<void(const RegisterState&)> callback)
+    : HardwareTimer(IRQ_TIMER, move(callback))
+{
+    InterruptDisabler disabler;
+    NonMaskableInterruptDisabler nmi_disabler;
+    enable_irq();
+    CMOS::write(0x8B, CMOS::read(0xB) | 0x40);
+    reset_to_default_ticks_per_second();
+}
+void RealTimeClock::handle_irq(const RegisterState& regs)
+{
+    HardwareTimer::handle_irq(regs);
+    CMOS::read(0x8C);
+}
+
+size_t RealTimeClock::ticks_per_second() const
+{
+    return m_frequency;
+}
+
+void RealTimeClock::reset_to_default_ticks_per_second()
+{
+    InterruptDisabler disabler;
+    ASSERT(try_to_set_frequency(1024));
```

Comment:
Code inside `ASSERT` invocations will not be compiled in non-debug builds.
This means that if we compile without `#define DEBUG`, we will not call `try_to_set_frequency()`